### ISSUE TITLE
Internal Documentation for JSON Forms components

### ISSUE THREAD

**Issue by CarlosNZ** [msupply-foundation/open-msupply#5380]
Been thinking we need this for a while. There are a lot of JSON Forms components and currently the only way to understand what they all do and how to configure them is to look at comments in the codebase. Would be good to have a really solid reference doc for all the components:
- what they do
- why we need them
- how they look (screenshots)
- configuration options

This would make configuring new forms easier, and reduce the likelihood of someone adding duplicate functionality because they didn't realise we it could be already done with an existing component (or by slightly amending an existing component).

I'm thinking something like the Form Element definitions wiki for Conforma: https://github.com/msupply-foundation/conforma-web-app/wiki/Element-Type-Specs

**Comment by andreievg** [comment]
@CarlosNZ you've done an amazing job with Conforma doc for this sort of thing, would be great if you have time to think of a doc template for omSupply custom components. Also would be great to have a commit example of adding a new custom component

**Comment by CarlosNZ** [comment]
> would be great if you have time to think of a doc template for omSupply custom components. Also would be great to have a commit example of adding a new custom component

Sure, I'll start one next week. 

Probably good to put this in the [Wiki](https://github.com/msupply-foundation/open-msupply/wiki), right?

**Comment by andreievg** [comment]
> Probably good to put this in the [Wiki](https://github.com/msupply-foundation/open-msupply/wiki), right?

Sounds like a good place !

**Comment by CarlosNZ** [comment]
> Although I kind of like the idea of version control and PRs for docs

Yeah, that's a good point. I just tested and you don't seem to be able to make PRs for the wiki repo :/

There's a README.md in the root of the programs package. I'll a new one in the JsonForms subfolder and link to it from the Programs one 👍🏼

**Comment by jmbrunskill** [comment]
Maybe add a link to the readme from the wiki so people can find it in both directions?

**Comment by roxy-dao** [comment]
Can I bump this. Can we move the milestone and actually assign someone?
